connectors/src/connectors/zendesk/temporal/activities.ts

```diff
@@ -284,50 +280,51 @@ export async function syncZendeskCategoryActivity({
   if (categoryInDb.permission === "none") {
     await deleteCategoryChildren({ connectorId, dataSourceConfig, categoryId });
     await categoryInDb.delete();
-    return null;
+    return false;
   }
 
-  const { accessToken, subdomain } = await getZendeskSubdomainAndAccessToken(
-    connector.connectionId
+  const zendeskApiClient = createZendeskClient(
+    await getZendeskSubdomainAndAccessToken(connector.connectionId)
   );
-  const zendeskApiClient = createZendeskClient({
-    token: accessToken,
-    subdomain,
+  await changeZendeskClientSubdomain(zendeskApiClient, {
+    connectorId,
+    brandId,
   });
-  await changeZendeskClientSubdomain({ client: zendeskApiClient, brandId });
 
   // if the category is not on Zendesk anymore, we delete it
   const { result: fetchedCategory } =
     await zendeskApiClient.helpcenter.categories.show(categoryId);
   if (!fetchedCategory) {
     await deleteCategoryChildren({ connectorId, categoryId, dataSourceConfig });
     await categoryInDb.delete();
-    return null;
+    return false;
   }
 
   // otherwise, we update the category name and lastUpsertedTs
   await categoryInDb.update({
     name: fetchedCategory.name || "Category",
     lastUpsertedTs: new Date(currentSyncDateMs),
   });
-  return categoryInDb;
+  return true;
 }
 
 /**
- * This activity is responsible for syncing all the articles in a Category.
+ * This activity is responsible for syncing the next batch of articles to process.
  * It does not sync the Category, only the Articles.
  */
-export async function syncZendeskArticlesActivity({
+export async function syncZendeskArticleBatchActivity({
   connectorId,
-  category,
+  categoryId,
   currentSyncDateMs,
   forceResync,
+  cursor,
 }: {
   connectorId: ModelId;
-  category: ZendeskCategoryResource;
+  categoryId: number;
```

Comment:
the goal here was to avoid leaking user data in Temporal workflows https://github.com/dust-tt/dust/pull/8450
This does add the need for an extra db fetch though